lib/Parser/Parser.yy

```diff
@@ -632,23 +810,53 @@ unique_shared_opt:
 
 literal:
       boolean_literal
-    | intLit
-    | floatLit
-    | stringLit
+    | intLit { 
+          int intVal = std::stoi($1.getLexeme().str());
+          $$ = ctx.getASTMemoryArena().create<LiteralExpr>(
+                      llvm::APInt(32, intVal),
+                      ctx.getTypesMemoryArena().create<IntTy>(IntTy::Signed, 32),
```

Comment:
these should probably be type variables with an int literal constraint, we don't know the exact type yet